replace mouse and touch events to pointer event to fix issue #8200

Oxc3 opened the pull request:
## What does the pull request do?
merges mouse and touch events into pointer events


## What is the current behavior?
on touch devices both mouse and touch events are produced by the browser, then Avalonia produces 2 click events


## What is the updated/expected behavior with this PR?
on touch devices a single click event

## Fixed issues
Fixes #8200

Oxc3 committed:
replace mouse and touch events to pointer event to fix issue #8200

dnfadmin commented:
[![CLA assistant check](https://cla.dotnetfoundation.org/pull/badge/not_signed)](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=8234) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=8234) before we can accept your contribution.<br/><br/>:x: Oxc3 [sign now](https://cla.dotnetfoundation.org/AvaloniaUI/Avalonia?pullRequest=8234)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/AvaloniaUI/Avalonia?pullRequest=8234) it.</sub>

maxkatz6 commented on src/Web/Avalonia.Web.Blazor/AvaloniaView.razor.cs:
```diff
@@ -56,90 +56,91 @@ internal INativeControlHostImpl GetNativeControlHostImpl()
+        private void OnPointerMove(Microsoft.AspNetCore.Components.Web.PointerEventArgs e)
         {
-            foreach (var touch in e.ChangedTouches)
+            if (e.PointerType == "mouse")
```
Let's change here and in other places to this:
```suggestion
            if (e.PointerType != "touch")
```

Because I tested with my pen tablet, PointerType comes as "pen". And it's not handled with neither of these events when we filter only by mouse or touch.

maxkatz6 replied:
For now, it's fine if pen will be handled in the same way as a mouse. There is no better pen API atm.

maxkatz6 commented on src/Web/Avalonia.Web.Blazor/AvaloniaView.razor.cs:
```diff
@@ -56,90 +56,91 @@ internal INativeControlHostImpl GetNativeControlHostImpl()
         }
-
-        private void OnTouchStart(TouchEventArgs e)
+                
```
Minor
```suggestion

```

Oxc3 committed:
Update AvaloniaView.razor.cs

modified pointer up, down, move to handle touch device specifically and all other devices as mouse event
